src/robothub_oak/live_view.py

```diff
@@ -0,0 +1,290 @@
+                "clickable": True
+            },
+            "detection": {
+                "thickness": 1,
+                "fill_transparency": 0.05,
+                "box_roundness": 0,
+                "color": [0, 255, 0],
+                "bbox_style": 0,
+                "line_width": 0.5,
+                "line_height": 0.5,
+                "hide_label": False,
+                "label_position": 0,
+                "label_padding": 10
+            },
+            'text': {
+                'font_color': [255, 255, 0],
+                'font_transparency': 0.5,
+                'font_scale': 1.0,
+                'font_thickness': 2,
+                'bg_transparency': 0.5,
+                'bg_color': [0, 0, 0]
+            }
+        },
+        "objects": [
+            {
+                "type": "detections",
+                "detections": []
+            }
+        ]
+    }
+
+    # Bounding boxes
+    for roi, label in zip(rectangles, rectangle_labels):
+        xmin, ymin, xmax, ymax = roi
+        metadata['objects'][0]['detections'].append(
+            {'bbox': [xmin, ymin, xmax, ymax], 'label': label, 'color': [0, 255, 255]}
+        )
+
+    # Texts
+    for text in texts:
+        metadata["objects"].append(text.prepare().serialize())
+
+    # Lines
+    for line in lines:
+        metadata["objects"].append(line.prepare().serialize())
+
+    # Publish
+    stream_handle.publish_video_data(bytes(h264_frame), timestamp, metadata)
+
+
+class LiveView:
+    def __init__(self,
+                 title: str,
+                 unique_key: str,
+                 device_mxid: str,
+                 frame_width: int,
+                 frame_height: int):
+        """
+        :param title: Name of the Live View.
+        :param unique_key: Live View identifier.
+        :param device_mxid: MXID of the device that is streaming the Live View.
+        :param frame_width: Frame width.
+        :param frame_height: Frame height.
+        """
+        self.frame_width = frame_width
+        self.frame_height = frame_height
+        self.device_mxid = device_mxid
+        self.unique_key = unique_key
+        self.title = title
+
+        self.stream_handle = _create_stream_handle(camera_serial=device_mxid, unique_key=unique_key, title=title)
+
+        # Objects
+        self.texts: List[VisText] = []
+        self.rectangles: List[BoundingBox] = []
+        self.labels: List[str] = []
+        self.lines: List[VisLine] = []
+
+    @staticmethod
+    def create(device: OakCamera,
+               component: Component,
+               title: str,
+               unique_key: str = None,
+               manual_publish: bool = False
+               ) -> 'LiveView':
+        """
+        Creates a Live View for a given component.
+
+        :param device: OakCamera instance.
+        :param component: Component to create a Live View for. Either a CameraComponent, StereoComponent or NNComponent.
+        :param title: Name of the Live View.
```

Comment:
title is not a good name because it needs an explanation
`:param title: Name of the Live View.`
lets call it `name`